packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+    if (this.curve == 'Ed25519') {
+      return new PrivateKey.Ed25519(privateKeyBytes)
+    } else {
+      return new PrivateKey.Secp256k1(privateKeyBytes)
+    }
+  }
+
+  /**
+   * Get the address for a given account.
+   * @deprecated Use getAddressAsync instead.
+   */
+  getAddress(): string {
+    throw new Error('getAddress is synchronous and cannot retrieve addresses directly. Use getAddressAsync instead.')
+  }
+
+  /**
+   * Get the current address asynchronously for a given account.
+   * @returns {Address} A promise resolving to the current address.
+   * A phrase is needed to create a wallet and to derive an address from it.
+   */
+  async getAddressAsync(): Promise<string> {
+    const networkId = this.getRadixNetwork()
+    const radixPrivateKey = this.getRadixPrivateKey()
+    const address = await LTSRadixEngineToolkit.Derive.virtualAccountAddress(radixPrivateKey.publicKey(), networkId)
+    return address.toString()
+  }
+
+  /**
+   * Get the explorer URL based on the network.
+   *
+   * @returns {string} The explorer URL based on the network.
+   */
+  getExplorerUrl(): string {
+    switch (this.getRadixNetwork()) {
+      case NetworkId.Mainnet:
+        return 'https://dashboard.radixdlt.com'
+      case NetworkId.Stokenet:
+        return 'https://stokenet-dashboard.radixdlt.com'
+      default:
+        throw new Error('Unsupported network')
+    }
+  }
+
+  /**
+   * Get the Gateway URL based on the network.
+   *
+   * @returns {string} The explorer URL based on the network.
+   */
+  getGatewayUrl(): string {
+    switch (this.getRadixNetwork()) {
+      case NetworkId.Mainnet:
+        return MAINNET_GATEWAY_URL
+      case NetworkId.Stokenet:
+        return STOKENET_GATEWAY_URL
+      default:
+        throw new Error('Unsupported network')
+    }
+  }
+
+  /**
+   * Get the explorer URL for a given account address based on the network.
+   * @param {Address} address The address to generate the explorer URL for.
+   * @returns {string} The explorer URL for the given address.
+   */
+  getExplorerAddressUrl(address: Address): string {
+    return `${this.getExplorerUrl()}/account/${address}`
+  }
+
+  /**
+   * Get the explorer URL for a given transaction ID based on the network.
+   * @param {string} txID The transaction ID to generate the explorer URL for.
+   * @returns {string} The explorer URL for the given transaction ID.
+   */
+  getExplorerTxUrl(txID: string): string {
+    return `${this.getExplorerUrl()}/transaction/${txID}`
+  }
+
+  /**
+   *  Validate the given address.
+   * @param {Address} address The address to validate.
+   * @returns {boolean} `true` if the address is valid, `false` otherwise.
+   */
+  async validateAddressAsync(address: string): Promise<boolean> {
+    try {
+      await RadixEngineToolkit.Address.decode(address)
+      return true
+    } catch (error) {
+      return false
+    }
+  }
+
+  validateAddress(): boolean {
```

Comment:
I think that we might want to provide this without using the toolkit. We could:

1. Ensure that the address can be Bech32m decoded.
2. Ensure that the HRP and that the entity byte match.
3. Ensure that the entity byte is an allowed one.
4. Ensure that the length is valid.